tests/ga/extension_actor.py

```diff
@@ -0,0 +1,465 @@
+        # haven't satisfied the http_get request. Returning None here does that.
+        return None
+    
+    return mock_http_get
+
+def _generate_mock_http_put(actors):
+
+    def http_put_record_status(url, *args, **kwargs):
+        if HttpRequestPredicates.is_host_plugin_status_request(url):
+            return None
+
+        handlerStatuses = json.loads(args[0]).get('aggregateStatus', {}).get('handlerAggregateStatus', [])
+
+        for handlerStatus in handlerStatuses:
+            supplied_name = handlerStatus.get('handlerName', None)
+            supplied_version = handlerStatus.get('handlerVersion', None)
+            
+            try: 
+                matches_info = lambda actor: \
+                    actor.extension_info.name == supplied_name \
+                        and actor.extension_info.version == supplied_version
+                
+                next(
+                    actor for actor in actors
+                    if matches_info(actor)
+                ).statusBlobs.append(handlerStatus)
+
+            except StopIteration as e:
+                # Tests will want to know that the agent is running an extension they didn't specifically allocate.
+                raise Exception("Status submitted for non-emulated extension: {0}".format(json.dumps(handlerStatus)))
+
+    return http_put_record_status
+
+
+@contextlib.contextmanager
+def get_protocol_and_handler(firstActor, *remainingActors):
+    """
+    A wrapper for mockwiredata.mock_wire_protocol() that adapts a set of ExtensionActor objects into inputs that 
+    that function can work with, whilst also creating and returning an exthandlers_handler instance pointing to the
+    returned protocol. Additionally, Popen is patched (for the scope of the with) to enable the returned exthandlers_handler 
+    to interact with every extension actor supplied. Only firstActor need be populated.
+
+    Note that non-default manifest.xml files must use the same url format as the default in order to be compatible with this
+    function. Specifically, the *.zip resource urls for an extension must end in "{ExtensionName}__{ExtensionVersion}".
+    """
+
+    with mock_wire_protocol(firstActor.data_fetcher, mockwiredata_factory=mockwiredata.get_dynamic_wire_protocol_data) as protocol:
+        
+        # We save the patched popens within the protocol to enable additions in future function calls
+        # (e.g. update_extension_actors).
+        protocol._patched_popens = [ ] # Populated later; timing matters, and a patched_popen needs to be started immediately after creation.
+        # We also need actor objs to, amoung other things, guarantee that the same actor doesn't have
+        # two different patch_popen's running at the same time.
+        protocol._actors = [firstActor]
+        protocol._actors.extend(remainingActors)
+
+        protocol.set_http_handlers(http_get_handler=_generate_mock_http_get(protocol._actors[1:]),
```

Comment:
bump

Reply:
Weirdly, this first comment doesn't show up in the Conversation's tab for me, even if "bump" does... Anyways, this code probably needs a comment; the reasoning here is that we're instantiating the `WireProtocol` object with the `first_actor`'s data (line 198: `first_actor.data_fetcher`), so that data gets queried after the `http_get_handler` returns, assuming that said handler doesn't actually handle the request. 

I suppose that we could add the first actor to the handler: it would be redundant, but wouldn't (shouldn't) change the overall effect.